### PR TITLE
make secrets configurable

### PR DESCRIPTION
**Pull request by al8n**

### CODE REVIEW

File: crates/core/src/config/secret.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    pub fn cipher(&self) -> XChaCha20Poly1305 {
+        XChaCha20Poly1305::new(GenericArray::from_slice(&self.cipher))
+    }
+
+    #[inline]
+    pub fn transport_keypair(&self) -> &TransportKeypair {
+        &self.transport_keypair
+    }
+
+    pub(super) fn persist(&self) -> std::io::Result<()> {
```
Comment (iduartgomez) [inline crates/core/src/config/secret.rs:173]: we don't need this, we should never be persisting those, either they are passed to the app by argument or we use default ones (or generate o. the flight for the transport key, every new execution can use a different key)

File: crates/core/src/config/secret.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+    let pk = rsa::RsaPrivateKey::from_pkcs1_pem(&buf).map_err(|e| {
+        std::io::Error::new(
+            std::io::ErrorKind::InvalidData,
+            format!("Failed to read key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(TransportKeypair::from_private_key(pk))
+}
```
Comment (iduartgomez) [inline crates/core/src/config/secret.rs:259]: add a unit test for loading the from disk or using default ones to see it works correctly. you can create the necessary files on a temp dir etc.

File: crates/core/src/wasm_runtime/secrets_store.rs [hunk 3]
```diff
@@ -94,21 +97,21 @@ impl StoreFsManagement for SecretsStore {
     }
 }
 
-static DEFAULT_CIPHER: Lazy<XChaCha20Poly1305> = Lazy::new(|| {
-    let arr = GenericArray::from_slice(&DelegateRequest::DEFAULT_CIPHER);
-    XChaCha20Poly1305::new(arr)
-});
+// static DEFAULT_CIPHER: Lazy<XChaCha20Poly1305> = Lazy::new(|| {
```
Comment (iduartgomez) [inline crates/core/src/wasm_runtime/secrets_store.rs:100]: lets remove this comments if they are not being used anymore

File: crates/core/src/config/secret.rs [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        let (nonce_path, nonce) = if let Some(nonce) = nonce {
+            (self.nonce.unwrap(), nonce)
+        } else {
+            // try secret dir
+            let path = secrets_dir.join(NONCE_FILENAME);
+            if path.exists() {
+                let nonce = read_nonce(&path)?;
+                (path, nonce)
+            } else {
+                (path, DelegateRequest::DEFAULT_NONCE)
```
Comment (iduartgomez) [inline crates/core/src/config/secret.rs:85]: I suppose this requires updating the stdlib, or is already present there?

Comment (al8n) [reply crates/core/src/config/secret.rs:85]: It is already there.

File: crates/core/src/config/secret.rs [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    use super::*;
+
+    #[test]
+    fn test_load() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let secrets_dir = temp_dir.path().join("secrets");
+        std::fs::create_dir_all(&secrets_dir).unwrap();
+
+        let transport_keypair = TransportKeypair::new();
+        let nonce = DelegateRequest::DEFAULT_NONCE;
```
Comment (iduartgomez) [inline crates/core/src/config/secret.rs:249]: we want 2 test cases at least, one where we load default values, and an other where we load different values , and assert both cases work correctly

File: crates/core/src/config/secret.rs [hunk 6]
```diff
@@ -240,14 +240,14 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_load() {
+    fn test_load_from_secret_dir() {
```
Comment (iduartgomez) [inline crates/core/src/config/secret.rs:243]: I think you are mixing up 2 different things. Secret dir are not for storing key, but for storing delegate secrets which is different! (Those may be keys or whatever private information in general). The keys is what we generate/load/default to encrypt those secrets. We don't load any key from secret dir neither store keys in there.

Comment (iduartgomez) [reply crates/core/src/config/secret.rs:243]: isn't this test redundant with `test_load_from_different_files` anyways. let's just remove this and then is good

**Review by iduartgomez — APPROVED** [review]

**Pull request merged**